discovery/discovery.go

```diff
@@ -202,14 +205,13 @@ type FailedAttempt struct {
 	Error  error
 }
 
-func walker(out *Endpoints, attempts *[]FailedAttempt, testFn DiscoverWalkFunc) DiscoverWalkFunc {
-	return func(pre string, eps *Endpoints, err error) error {
+func walker(attempts *[]FailedAttempt, testFn DiscoverWalkFunc) DiscoverWalkFunc {
+	return func(pre string, dd *discoveryData, err error) error {
 		if err != nil {
 			*attempts = append(*attempts, FailedAttempt{pre, err})
 			return nil
 		}
-		out.Append(*eps)
```

Comment:
I can't get a sense on appending the endpoint or keys obtained in the child path to the ones obtained in this path since this should never happene because if I get an endpoint at a path I should consider discovery as done and return the result without walking up to the parent path.

Perhaps I'm missing something...